### DISCUSSION TITLE
Is the Infobar supposed to show somewhere by default?

### DISCUSSION THREAD

**Discussion by ronin712** [gildas-lormeau/SingleFile#1221] [Q&A]
The help for the setting "include the infobar in the saved page" says "Check this option to display the ℹ button at the top right of the page when viewing a saved page in a browser where SingleFile is not installed."

This seems to imply that if it's not checked but you have the extension installed you should see an infobar.  I don't see an infobar anywhere, and can only get to the infobar info if I turn this setting on and click the "i" button.

MacOS 13.4.1 (c), Safari 16.5.2 (18615.2.9.11.10), SingleFile 1.0.26

**Comment by gildas-lormeau** [comment]
Enable `Develop > Disable Local File Restrictions` in Safari. 

Alternatively, just wait some hours until the new version (1.0.28) is available on the store. I redesigned the infobar in order to work without needing JS to be enabled. It should be available for download very soon.

**Reply by gildas-lormeau** [reply]
The version 1.0.28 is now available for download on the store :)

**Reply by ronin712** [reply]
Thanks!

**Comment by ronin712** [comment]
I've tried both Disable Local File Restrictions and the new version.  I still don't see an info bar either in Safari (where the extension is installed) and Firefox (where it's not installed).

Also, the new version seems to have broken \n in the Infobar text (which I can only see when I use the "include the infobar in the saved page" setting).

**Comment by gildas-lormeau** [comment]
I confirm that the infobar will be visible in Safari **if and only if** the option `Infobar > include the infobar in the saved page` is enabled in SingleFile. However, since the version `1.0.28`, you don't need to enable  the option `Develop > Disable Local File Restrictions` in Safari (to enable the support of JavaScript) anymore.  I am sorry for the misunderstanding.

I was also able to reproduce and fix the issue regarding the support `\n` (and `\t`) when JavaScript is disabled (e.g. when displaying a page in Safari). The fix will be available in the next version.

**Reply by ronin712** [reply]
Sounds good, thank you!

**Reply by bennjmin** [reply]
I wonder why this choice was made, to work with javascript and an info button. This is not a complaint, I just discovered SingleFile, and I should have discovered it much earlier. The infobar works perfect for me. But I always prefer simplicity. I would have opted to just show this information on top of the page, in a simple box or something. And the url simply being clickable as an old fashioned link. No javascript involved at all.

**Reply by gildas-lormeau** [reply]
@bennjmin  Actually I improved the implementation of the infobar meanwhile. It should work with JavaScript disabled when the infobar is embedded in the saved page. For the record, here is a demo: https://plnkr.co/edit/dqGmg22rRiRnsgyd?preview.
